src/datatypes/interval.coffee

```diff
@@ -166,12 +166,16 @@ module.exports.Interval = class Interval
       null
 
   sameAs: (other, precision) ->
-    if ((@low == null and @high == null) or (other.low == null and other.high == null)) then return null
-    # pass in the opposite value to get the type since the checked value is null
-    else if @low == null then @low = minValueForInstance(@high)
-    else if other.low == null then other.low = minValueForInstance(other.high)
-    else if @high == null then @high = maxValueForInstance(@low)
-    else if other.high == null then other.high = maxValueForInstance(other.low)
+    # Checks to see if any of the Intervals have a open, null boundary, returns null if true
+    if (@low == null and !@lowClosed) or (@high == null and !@highClosed) or (other.low == null and !other.lowClosed) or (other.high == null and !other.highClosed) then return null
+
+    # For the special case where Interval[null,null] same as Interval[null,null] to return true
+    if @lowClosed and !@low? and @highClosed and !@high? then return other.lowClosed and !other.low? and other.highClosed and !other.high?
+
+    # For the special case where Interval[...] same as Interval[null,null] should return false
+    # This accounts for the inverse of the if statement above: where the second Interval is [null,null] and not the first Interval
+    if other.lowClosed and !other.low? and other.highClosed and !other.high? then return false
```

Comment:
When other.start() or other.end() is called, it is assumed that if the value is null and closed, that the opposite boundary is not null and it is passed to the Min or Max value function. It uses the opposite boundary to determine the datatype that is to be returned. Since it is null, it just returns null. There really isn't a good way to pass in a value since this calculation is done inside the interval. It has no knowledge of the interval it is being compared to. That is why I have an if block to catch that case.